src/main/scala/io/iohk/ethereum/jsonrpc/JsonRpcController.scala

```diff
@@ -0,0 +1,46 @@
+package io.iohk.ethereum.jsonrpc
+
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.Web3Service.{ClientVersionRequest, ClientVersionResponse, Sha3Request, Sha3Response}
+import org.json4s.JsonAST.JArray
+import org.json4s.JsonDSL._
+
+import scala.concurrent.Future
+import scala.util.Try
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object JsonRpcController {
+  trait JsonDecoder[T] {
+    def decodeJson(params: JArray): Try[T]
+  }
+
+  trait JsonEncoder[T] {
+    def encodeJson(t: T): JArray
+  }
+}
+
+class JsonRpcController(web3Service: Web3Service) {
+
+  import JsonMethodsImplicits._
+
+  def handleRequest(request: JsonRpcRequest): Future[JsonRpcResponse] = {
+    request.method match {
+      case "web3_sha3" => handle[Sha3Request, Sha3Response](web3Service.sha3, request)
+      case "web3_clientVersion" => handle[ClientVersionRequest, ClientVersionResponse](web3Service.clientVersion, request)
```

Comment:
It seems that this api call is failing if we don't send an empty array as parameters. In parity it works with or without it